### PR TITLE
Revive PR - Extend Service Bus with Active and Deadletter-Queue thresholds

### PR DESCRIPTION
**Pull request by cieciurm**
**What this PR does / why we need it**:
This PR revives #822

- Fixed build
- Fixed unit tests (and moved to specific AzureServiceBus project)
- Fixed registration

**Which issue(s) this PR fixes**:

Please reference the issue this PR will close: #821

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
No.

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [x] Created/updated tests
- [x] Unit tests passing
- [x] End-to-end tests passing
- [x] Extended the documentation
- [x] Provided sample for the feature

### CODE REVIEW

File: src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+using HealthChecks.AzureServiceBus.Configuration;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace HealthChecks.AzureServiceBus;
+
+public class AzureServiceBusQueueMessageCountThresholdHealthCheck : AzureServiceBusHealthCheck<AzureServiceBusQueueMessagesCountThresholdHealthCheckOptions>, IHealthCheck
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs:6]: I would merge that check with `AzureServiceBusDeadLetterQueueMessageCountThresholdHealthCheck ` into single one and play with options object to set all 4 thresholds:
```
public struct Threshold
{
public int Degraded {get;}
public int Unhealthy {get;}
}

public AzureServiceBusQueueMessagesCountThresholdHealthCheckOptions 
{
   public Threshold ActiveMessages {get; set; }
   public Threshold DeadLetterMessages { get; set; }
}
```

File: src/HealthChecks.AzureServiceBus/AzureServiceBusDeadLetterQueueMessageCountThresholdHealthCheck.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context,
+        CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            if (!ManagementClientConnections.TryGetValue(ConnectionKey, out var managementClient))
+            {
+                managementClient = CreateManagementClient();
+
+                if (!ManagementClientConnections.TryAdd(ConnectionKey, managementClient))
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/AzureServiceBusDeadLetterQueueMessageCountThresholdHealthCheck.cs:32]: Use `GetOrAdd` API instead.

File: src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs (hunk 1 above)
Comment (cieciurm) [reply src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs:7]: It's been changed

File: src/HealthChecks.AzureServiceBus/AzureServiceBusDeadLetterQueueMessageCountThresholdHealthCheck.cs (hunk 2 above)
Comment (cieciurm) [reply src/HealthChecks.AzureServiceBus/AzureServiceBusDeadLetterQueueMessageCountThresholdHealthCheck.cs]: Done

**Comment by cieciurm** [comment]
Hi,
Thanks for your feedback.
Changes applied.

cc @sungam3r

File: src/HealthChecks.AzureServiceBus/Configuration/AzureServiceBusQueueMessagesCountThresholdHealthCheckOptions.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+/// <summary>
+/// Configuration options for <see cref="AzureServiceBusQueueMessageCountThresholdHealthCheck"/>.
+/// </summary>
+public class AzureServiceBusQueueMessagesCountThresholdHealthCheckOptions : AzureServiceBusQueueHealthCheckOptions
+{
+    /// <summary>
+    /// Indicates if dead letter messages queue or active messages queue should be checked.
+    /// </summary>
+    public bool CheckDeadLetterMessages { get; }
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/Configuration/AzureServiceBusQueueMessagesCountThresholdHealthCheckOptions.cs:11]: Why not both?

Comment (sungam3r) [reply src/HealthChecks.AzureServiceBus/Configuration/AzureServiceBusQueueMessagesCountThresholdHealthCheckOptions.cs:11]: I mean why do you need that setting at all?

File: src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+        _degradedThreshold = threshold.DegradedThreshold;
+        _unhealthyThreshold = threshold.UnhealthyThreshold;
+    }
+
+    protected override string ConnectionKey => $"{Prefix}_{_queueName}";
+
+    /// <inheritdoc />
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context,
+        CancellationToken cancellationToken = default)
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs:30]:
```suggestion
    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
```

File: src/HealthChecks.AzureServiceBus/Configuration/AzureServiceBusQueueMessagesCountThresholdHealthCheckOptions.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+    /// <summary>
+    /// Threshold configuration for active messages queue.
+    /// </summary>
+    public AzureServiceBusQueueMessagesCountThreshold ActiveMessages { get; set; }
+
+    /// <summary>
+    /// Threshold configuration for dead letter messages queue.
+    /// </summary>
+    public AzureServiceBusQueueMessagesCountThreshold DeadLetterMessages { get; set; }
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/Configuration/AzureServiceBusQueueMessagesCountThresholdHealthCheckOptions.cs:21]: unused

File: src/HealthChecks.AzureServiceBus/Configuration/AzureServiceBusQueueMessagesCountThresholdHealthCheckOptions.cs (hunk 3 above)
Comment (sungam3r) [reply src/HealthChecks.AzureServiceBus/Configuration/AzureServiceBusQueueMessagesCountThresholdHealthCheckOptions.cs:11]: I suggest to check both types of messages (if corresponding threshold was specified) in a single health check.

**Comment by cieciurm** [comment]
Thanks @sungam3r!

I have merged into a single health check - the only down-side is I changed from `struct` to `class` because I wanted to use `Guard` clause.

Let me know what do you think now :)

Also I'm wondering if we want to add registration method which supports two `Func` to configure both active and dead letter messages

**Comment by sungam3r** [comment]
> from struct to class because I wanted to use Guard clause.

😕  For struct `Guard` clause is useless since structs always non-null, but I'm fine either class or struct.

File: src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+            var properties = await managementClient.GetQueueRuntimePropertiesAsync(_queueName, cancellationToken).ConfigureAwait(false);
+
+            var activeQueueHealthStatus = CheckHealthStatus(
+                properties,
+                x => x.ActiveMessageCount,
+                _activeMessagesThreshold,
+                "queue");
+
+            if (!IsHealthy(activeQueueHealthStatus))
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs:38]:
```suggestion
            if (activeQueueHealthStatus.Status != HealthStatus.Healthy)
```

File: src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+        if (messagesCount >= threshold.DegradedThreshold)
+        {
+            return HealthCheckResult.Degraded($"Message in {queueType} {_queueName} exceeded the amount of messages allowed for the degraded threshold {threshold.DegradedThreshold}/{messagesCount}");
+        }
+
+        return HealthCheckResult.Healthy();
+    }
+
+    private bool IsHealthy(HealthCheckResult result) => result.Status == HealthStatus.Healthy;
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs:88]:
```suggestion
```

File: src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+        }
+        catch (Exception ex)
+        {
+            return new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
+        }
+    }
+
+    private HealthCheckResult CheckHealthStatus(
+        QueueRuntimeProperties queueRuntimeProperty,
+        Func<QueueRuntimeProperties, long> messagesCountFunc,
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs:64]:
```suggestion
        long messagesCount,
```

File: src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+        AzureServiceBusQueueMessagesCountThreshold threshold,
+        string queueType)
+    {
+        if (threshold is null)
+        {
+            return HealthCheckResult.Healthy();
+        }
+
+        var messagesCount = messagesCountFunc(queueRuntimeProperty);
+
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs:74]:
```suggestion
```

File: src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+            return HealthCheckResult.Healthy();
+        }
+        catch (Exception ex)
+        {
+            return new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
+        }
+    }
+
+    private HealthCheckResult CheckHealthStatus(
+        QueueRuntimeProperties queueRuntimeProperty,
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs:63]:
```suggestion
```

File: src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+                "queue");
+
+            if (!IsHealthy(activeQueueHealthStatus))
+            {
+                return activeQueueHealthStatus;
+            }
+
+            var deadLetterQueueHealthStatus = CheckHealthStatus(
+                properties,
+                x => x.DeadLetterMessageCount,
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs:45]:
```suggestion
                properties.DeadLetterMessageCount,
```

File: src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        try
+        {
+            var managementClient = ManagementClientConnections.GetOrAdd(ConnectionKey, CreateManagementClient());
+
+            var properties = await managementClient.GetQueueRuntimePropertiesAsync(_queueName, cancellationToken).ConfigureAwait(false);
+
+            var activeQueueHealthStatus = CheckHealthStatus(
+                properties,
+                x => x.ActiveMessageCount,
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs:34]:
```suggestion
                properties.ActiveMessageCount,
```

**Comment by sungam3r** [comment]
> if we want to add registration method which supports two Func to configure both active and dead letter messages

I would not bother with many overloads (4 new now) and just provide minimum - 1 new will be enough as I understand. All optional agruments can be set via options delegate. Have too many overloads is a burden to support.

File: src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs [hunk 13]
```diff
@@ -0,0 +1,82 @@
+using Azure.Messaging.ServiceBus.Administration;
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/AzureServiceBusQueueMessageCountThresholdHealthCheck.cs:1]:
```suggestion
```

File: src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs [hunk 14]
```diff
@@ -9,14 +9,17 @@ namespace Microsoft.Extensions.DependencyInjection;
...
+/// <see cref="AzureServiceBusSubscriptionHealthCheck"/>, <see cref="AzureServiceBusTopicHealthCheck"/>,
+/// <see cref="AzureServiceBusQueueMessageCountThresholdHealthCheck"/>.
 /// </summary>
 public static class AzureServiceBusHealthCheckBuilderExtensions
 {
     private const string AZUREEVENTHUB_NAME = "azureeventhub";
     private const string AZUREQUEUE_NAME = "azurequeue";
     private const string AZURETOPIC_NAME = "azuretopic";
     private const string AZURESUBSCRIPTION_NAME = "azuresubscription";
+    private const string AZUREQUEUETHRESHOLD_NAME = "azurequeuethreshold";
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs:21]: Only one name will be enough.

Comment (cieciurm) [reply src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs:21]: Done, thx

**Comment by cieciurm** [comment]
Thanks for your review :)

I decided to stick with `struct` - let me know what do you think now :)

**Comment by sungam3r** [comment]
When HC provides options object client can specify any configuration in action delegate so no need to provide 100500 extension methods from our side. Imagine you have options object with 10 optional properties inside. Would you provide all combinations for setting thoise properties? Obviously no.

**Comment by sungam3r** [comment]
A lot of extensions were added historucally **before** introducing options pattern. I personally dislike those bunch overloads but I did not remove already written ones. For new health checks I do not bother with many extension methods leaning to provide only required minimum - one with `Action<BlaBlaOptions>`. Anyway - eventually the number grows because of requirements to access to things from DI, so new overloads apprear with `IServiceProvider`, etc.

**Comment by cieciurm** [comment]
Thanks for the explanation, @sungam3r.

I got fixated to pass `Action<Threshold>` instead of `Action<Options>` to the extension method, and that is why I thought that you need extension method per threshold type.

Now, as you mentioned, it's a free/open to use as possible, because we expose `Action<Options>`.
There are two extension methods in total because there are two ways to authenticate:
- using Connection String
- using Endpoint and Token Credential

Tests are fixed as well :)

File: test/HealthChecks.AzureServiceBus.Tests/DependencyInjection/AzureServiceBusDeadLetterQueueMessageThresholdCountUnitWithTokenUnitTests.cs [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+public class azure_service_bus_deadletter_queue_message_threshold_registration_with_token_should
+{
+    [Fact]
+    public void add_health_check_when_properly_configured()
+    {
+        var services = new ServiceCollection();
+        services.AddHealthChecks()
+            .AddAzureServiceBusDeadLetterQueueMessageCountThreshold("cnn", "queueName", new MockTokenCredentials());
+
+        var serviceProvider = services.BuildServiceProvider();
```
[the remark this answers is not in the document]
Comment (cieciurm) [reply test/HealthChecks.AzureServiceBus.Tests/DependencyInjection/AzureServiceBusDeadLetterQueueMessageThresholdCountUnitWithTokenUnitTests.cs]: It's been applied

**Comment by sungam3r** [comment]
> There are two extension methods in total because there are two ways to authenticate

OK, I'm fine to provide 2, but again - these auth combinations are in options. What if you have 5 ways of auth? Would you provide 5 extension methods to connect your HC? What prompted you to choose a method of authentication as a criterion for the number of overloads? All these settings are optional and can generate combinations, especially with further evolution. Subsequently, this can create a design problem when the previously chosen criterion will no longer be so obvious and will have to change the public API.

**Comment by sungam3r** [comment]
In other words `string queueName` is good parameter for extension method because it's a required one to build `AzureServiceBusQueueMessagesCountThresholdHealthCheckOptions` (and then pass it to delegate back to the caller). All other parameters are not so good to enumerate in extension methods.

File: src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs [hunk 16]
```diff
@@ -384,6 +386,96 @@ public static class AzureServiceBusHealthCheckBuilderExtensions
             timeout));
     }
 
+    /// <summary>
+    /// Add a health check for specified Azure Service Bus Queue active or dead letter messages threshold
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs:390]:
```suggestion
    /// Add a health check for specified Azure Service Bus Queue active or dead letter messages threshold.
```

File: src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs [hunk 17]
```diff
@@ -384,6 +386,96 @@ public static class AzureServiceBusHealthCheckBuilderExtensions
...
+    /// </summary>
+    /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+    /// <param name="connectionString">The azure service bus connection string to be used.</param>
+    /// <param name="queueName">The name of the queue to check.</param>
+    /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'azurequeue' will be used for the name.</param>
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs:395]:
```suggestion
    /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'azurequeuethreshold' will be used for the name.</param>
```

File: src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs [hunk 18]
```diff
@@ -384,6 +386,96 @@ public static class AzureServiceBusHealthCheckBuilderExtensions
...
+        return builder.Add(new HealthCheckRegistration(
+            name ?? AZUREQUEUETHRESHOLD_NAME,
+            sp => new AzureServiceBusQueueMessageCountThresholdHealthCheck(options),
+            failureStatus,
+            tags,
+            timeout));
+    }
+
+    /// <summary>
+    /// Add a health check for specified Azure Service Bus Queue active or dead letter messages threshold
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs:433]:
```suggestion
    /// Add a health check for specified Azure Service Bus Queue active or dead letter messages threshold.
```

File: src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs [hunk 19]
```diff
@@ -384,6 +386,96 @@ public static class AzureServiceBusHealthCheckBuilderExtensions
...
+    /// </summary>
+    /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+    /// <param name="endpoint">The azure service bus endpoint to be used, format sb://myservicebus.servicebus.windows.net/.</param>
+    /// <param name="queueName">The name of the queue to check.</param>
+    /// <param name="tokenCredential">The token credential for auth</param>
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs:438]:
```suggestion
    /// <param name="tokenCredential">The token credential for authentication.</param>
```

File: src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs [hunk 20]
```diff
@@ -384,6 +386,96 @@ public static class AzureServiceBusHealthCheckBuilderExtensions
...
+    /// <param name="configure">An optional action to allow additional Azure Service Bus configuration.</param>
+    /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'azurequeue' will be used for the name.</param>
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs:440]:
```suggestion
    /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'azurequeuethreshold' will be used for the name.</param>
```

File: src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs [hunk 21]
```diff
@@ -384,6 +386,96 @@ public static class AzureServiceBusHealthCheckBuilderExtensions
...
+    /// <param name="connectionString">The azure service bus connection string to be used.</param>
+    /// <param name="queueName">The name of the queue to check.</param>
+    /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'azurequeue' will be used for the name.</param>
+    /// <param name="configure">An optional action to allow additional Azure Service Bus configuration.</param>
+    /// <param name="failureStatus">
+    /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <c>null</c> then
+    /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
+    /// </param>
+    /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+    /// <param name="timeout">An optional System.TimeSpan representing the timeout of the check.</param>
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs:402]:
```suggestion
    /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
```

File: src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs [hunk 22]
```diff
@@ -384,6 +386,96 @@ public static class AzureServiceBusHealthCheckBuilderExtensions
...
+    /// <param name="queueName">The name of the queue to check.</param>
+    /// <param name="tokenCredential">The token credential for auth</param>
+    /// <param name="configure">An optional action to allow additional Azure Service Bus configuration.</param>
+    /// <param name="name">The health check name. Optional. If <c>null</c> the type name 'azurequeue' will be used for the name.</param>
+    /// <param name="failureStatus">
+    /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <c>null</c> then
+    /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
+    /// </param>
+    /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+    /// <param name="timeout">An optional System.TimeSpan representing the timeout of the check.</param>
```
Comment (sungam3r) [inline src/HealthChecks.AzureServiceBus/DependencyInjection/AzureServiceBusHealthCheckBuilderExtensions.cs:446]:
```suggestion
    /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
```

**Review by sungam3r — APPROVED** [review]

**Comment by sungam3r** [comment]
@cieciurm I'll wait for your decision on the number of overloads until tomorrow.

**Comment by cieciurm** [comment]
I am no expert, but I think my pattern of thinking, as well as original PR author's, were inspired by the shape of existing extension methods for Queues.

If you see the current overloads of `AddAzureServicBusQueue` extension methods, they are embracing two different ways of authentication. The auth parameters are passed explicitly as parameters, and not thru `Action<Options>`.

I would rely on your expertise and experience in shaping the public API of a library - if you think it is just causing noise, then let's remove it and have single method.

Just let me know what you think and I will adjust the PR tomorrow morning :)

**Comment by sungam3r** [comment]
I merge as is. I still think that the library (almost each HC project) has more extensions to add health checks than necessary, but now is not the time to change the design removing that "noise".

**Pull request merged**